src/abstractdataframe/reshape.jl

```diff
@@ -399,3 +399,107 @@ function CategoricalArrays.CategoricalArray(v::RepeatedVector)
     res.refs = repeat(res.refs, inner = [v.inner], outer = [v.outer])
     res
 end
+
+
+Base.transpose(::AbstractDataFrame, args...; kwargs...) =
+    MethodError("`transpose` not defined for `AbstractDataFrame`s. Try `permutedims` instead")
+
+"""
+    permutedims(df::AbstractDataFrame [, src_namescol::Union{Int, Symbol, <:AbstractString}
+                    [, dest_namescol::Union{Symbol, AbstractString} ]];
+                    makeunique::Bool=false)
```

Comment:
Align with `df`.

Reply:
Ugh - a bunch of stuff seems like it got reverted after rebase somehow. Sorry about that